projects/cdk/observables/zone.ts

```diff
@@ -23,3 +28,32 @@ export function tuiZonefree<T>(zone: NgZone): MonoTypeOperatorFunction<T> {
 export function tuiZoneOptimized<T>(zone: NgZone): MonoTypeOperatorFunction<T> {
     return pipe(tuiZonefree(zone), tuiZonefull(zone));
 }
+
+class TuiZoneScheduler implements SchedulerLike {
+    constructor(
+        private readonly zoneConditionFn: <T>(fn: (...args: unknown[]) => T) => T,
+        private readonly scheduler: SchedulerLike = asyncScheduler,
+    ) {}
+
+    public now(): number {
+        return this.scheduler.now();
+    }
+
+    public schedule(...args: Parameters<SchedulerLike['schedule']>): Subscription {
+        return this.zoneConditionFn(() => this.scheduler.schedule(...args));
+    }
+}
+
+export function tuiZonefreeScheduler(
+    zone: NgZone,
+    scheduler: SchedulerLike = asyncScheduler,
+): SchedulerLike {
+    return new TuiZoneScheduler(zone.runOutsideAngular.bind(zone), scheduler);
+}
+
+export function tuiZonefullScheduler(
+    zone: NgZone,
```

Comment:
Let's write `= inject(NgZone)` here and in zone optimized functions so that we can just not pass it at place when they are in injection context